src/hooks/use-has-user-interacted.ts

```diff
@@ -0,0 +1,16 @@
+import { useContext, useEffect, useState } from 'react'
+
+import { UsercentricsContext } from '../context.js'
+import { hasUserInteracted } from '../utils.js'
+
+/** Returns `true` if user has interacted with the dialog. */
+export const useHasUserInteracted = (): boolean => {
+    const { ping } = useContext(UsercentricsContext)
+    const [userInteracted, setUserInteracted] = useState(hasUserInteracted())
+
+    useEffect(() => {
+        setUserInteracted(hasUserInteracted())
+    }, [ping])
+
+    return userInteracted
```

Comment:
It would be simpler to use `useMemo` here, which is basically the same thing under the hood:

```suggestion:-5
  return useMemo(() => hasUserInteracted(), [ping])
```

This will cause the value to be recalculated whenever `ping` updates.